pulpcore/tests/functional/api/test_crd_artifacts.py

```diff
@@ -21,26 +21,30 @@ def pulpcore_random_file(tmp_path):
     return {"name": name, "size": 1024, "digest": digest}
 
 
-def _do_upload_valid_attrs(artifact_api, file, data):
-    """Upload a file with the given attributes."""
-    artifact = artifact_api.create(file, **data)
-    # assumes ALLOWED_CONTENT_CHECKSUMS does NOT contain "md5"
-    assert artifact.md5 is None, "MD5 {}".format(artifact.md5)
-    read_artifact = artifact_api.read(artifact.pulp_href)
-    # assumes ALLOWED_CONTENT_CHECKSUMS does NOT contain "md5"
-    assert read_artifact.md5 is None
-    for key, val in artifact.to_dict().items():
-        assert getattr(read_artifact, key) == val
-    # Delete the Artifact
-    artifact_api.delete(read_artifact.pulp_href)
-    with pytest.raises(ApiException) as e:
-        artifact_api.read(read_artifact.pulp_href)
-
-    assert e.value.status == 404
+@pytest.fixture
+def upload_valid_attrs(allow_admin_destroy_artifact):
+    def _upload_valid_attrs(artifact_api, file, data):
```

Comment:
Now that this is a fixture, we can ask for the `pulpcore-binding` fixture:
```suggestion
@pytest.fixture
def upload_valid_attrs(pulpcore_bindings, allow_admin_destroy_artifact):
    def _upload_valid_attrs(file, data):
```